### PR TITLE
update openlibm dep for julia versions

### PR DESCRIPTION
**Pull request by ngam**
I have no idea what I am doing. 

We want to fix this: https://github.com/conda-forge/julia-feedstock/issues/169

This is not exactly a replace_pin situation, we just need to add a pin to openlibm... but I copied from the one above it.

```            
_replace_pin("openlibm", "openlibm <0.8.0", deps, record)
```

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -1234,7 +1234,7 @@ def _gen_new_index(repodata, subdir):
             
         # Retroactively pin a max version of openlibm for julia 1.6.* and 1.7.*:
         # https://github.com/conda-forge/julia-feedstock/issues/169
-        if record_name == "sphinx" and (record["version"].startswith("1.6") or record["version"].startswith("1.7")):
+        if record_name == "julia" and (record["version"].startswith("1.6") or record["version"].startswith("1.7")):
```
Comment (jaimergp) [inline recipe/gen_patch_json.py:1237]: `str.startswith` can take a tuple of `str` and returns `True` if any of those is true.

```suggestion
        if record_name == "julia" and record["version"].startswith(("1.6", "1.7")):
```

Comment (jaimergp) [reply recipe/gen_patch_json.py:1237]: Also, we might want to add a timestamp check too, so future packages (hopefully corrected) are not affected.

Comment (ngam) [reply recipe/gen_patch_json.py]: @jaimergp pls check timestamp addition, thanks

**Comment by jaimergp** [comment]
Looks good. Can you post the diff please? See [here](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/tree/master/recipe).

**Comment by ngam** [comment]
```diff
linux-64::julia-1.6.1-h96f571d_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h438a1c7_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h96f571d_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-hbb0e23a_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h2a39549_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h438a1c7_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h2a39549_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_2.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h989b2f6_2.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.0-h989b2f6_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h620718c_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.1-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-ha300950_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_2.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.0-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h1f75604_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
```

**Comment by ngam** [comment]
Before you merge, let's get an opinion from @mkitti and @conda-forge/julia in the next few hours.

Specifically, I chose to apply this for only julia 1.6.x and julia 1.7.x as they are the only versions supported. I intentionally left out the rest, which are likely will be rendered useless (and were useless anyway). Should I extend this to all versions?

**Comment by mkitti** [comment]
This looks fine to me. I have no idea about julia outside the 1.6.x and 1.7.x branches either, so looks good to me.

**Comment by ngam** [comment]
I don't see curl in the show_diff...

**Comment by mkitti** [comment]
Also it should really be libcurl I think

**Comment by ngam** [comment]
> Also it should really be libcurl I think

No, we use curl: https://github.com/conda-forge/julia-feedstock/blob/54eccbda430336c24b2b79fc9422e89857037547/recipe/meta.yaml#L46

**Comment by ngam** [comment]
@jaimergp any idea why curl doesn't show up in the show_diff? Should we just open another PR to do that?

**Comment by ngam** [comment]
Nvm. curl is pinned... fixing soon

**Comment by ngam** [comment]
```
python show_diff.py --use-cache
linux-64::julia-1.6.1-h96f571d_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h438a1c7_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h96f571d_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-hbb0e23a_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h2a39549_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h438a1c7_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h2a39549_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h989b2f6_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
linux-64::julia-1.7.0-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h620718c_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
osx-64::julia-1.6.1-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-ha300950_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
osx-64::julia-1.7.0-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.70.0,<7.80.0",
osx-64::julia-1.7.1-h1f75604_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",

```

**Comment by ngam** [comment]
```diff
python show_diff.py --use-cache
linux-64::julia-1.6.1-h96f571d_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h438a1c7_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h96f571d_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-hbb0e23a_1.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h2a39549_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h438a1c7_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h2a39549_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h989b2f6_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
linux-64::julia-1.7.0-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h620718c_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
osx-64::julia-1.6.1-h132cb31_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-ha300950_1.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
osx-64::julia-1.7.0-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
osx-64::julia-1.7.1-h1f75604_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",

```

**Comment by ngam** [comment]
```diff
linux-64::julia-1.6.1-h96f571d_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h438a1c7_3.tar.bz2
-    "curl >=7.79.1,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h96f571d_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-hbb0e23a_1.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h2a39549_1.tar.bz2
-    "curl >=7.79.1,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h438a1c7_0.tar.bz2
-    "curl >=7.79.1,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h2a39549_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h989b2f6_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
linux-64::julia-1.7.0-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h620718c_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
osx-64::julia-1.6.1-h132cb31_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_0.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_3.tar.bz2
-    "curl >=7.79.1,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-ha300950_1.tar.bz2
-    "curl >=7.78.0,<8.0a0",
+    "curl >=7.78.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_0.tar.bz2
-    "curl >=7.79.1,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_1.tar.bz2
-    "curl >=7.79.1,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
osx-64::julia-1.7.0-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_0.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_1.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_2.tar.bz2
-    "curl >=7.80.0,<8.0a0",
+    "curl >=7.80.0,<7.81.0",
osx-64::julia-1.7.1-h1f75604_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",

```

**Comment by jaimergp** [comment]
I feel this `curl` pin is too restrictive, since it seems to be caused by a bug, not a deliberate change in the ABI. If we limit Julia to this narrow interval (basically `=7.80.*`), all packages in the environment will have to use that one too. Maybe what we want to do instead is skipping `7.81.0` or `7.81.*`? The version range would be `>=7.80.0,!=7.81.0` in that case. I'd like @conda-forge/core to confirm though.

**Comment by beckermr** [comment]
The correct pin depends on Julia itself. If we think 7.82 will work, then skipping seems fine. If Julia is just going to break that too, then we might want to do more extreme things like statically link curl at build time.

**Comment by ngam** [comment]
> `>=7.80.0,!=7.81.0,<8.0a0` in that case

I like this (I would have gone for it if I new I could do it!)

**Comment by ngam** [comment]
> The correct pin depends on Julia itself. If we think 7.82 will work, then skipping seems fine. If Julia is just going to break that too, then we might want to do more extreme things like statically link curl at build time.

@beckermr how bad would it be if we manually modify global pins in julia? I was thinking if imposing strict upper bounds on all dependencies (not just curl and openlibm). Many of these deps have global pins and so that's making me pause to think. See my comment here: https://github.com/conda-forge/julia-feedstock/issues/172#issuecomment-1007625764

We are also thinking of building a `julia-jc` version (basically vendoring everything from upstream (jc: julia computing) or repackaging stuff) as a stable solution, and work on a `julia-cf` version (continue the current approach with more rigorous testing and evaluation of dependencies added)

**Comment by beckermr** [comment]
Changing the global pins is quite destructive. It will be a lot harder to solve environments.

We cannot vendor .so's since they will stomp on one another. 

We could link some deps statically into julia. This is not ideal but might ease burdens in other ways.

**Comment by beckermr** [comment]
cc @conda-forge/core

**Comment by isuruf** [comment]
Linking statically is not an option. Julia expects some dependencies to be shared libraries and do dlopen shared libraries.

**Comment by beckermr** [comment]
Ouch. Thanks for the info.

**Comment by mkitti** [comment]
We can just vendor curl as a private library at `$CONDA_PREFIX/lib/julia/libcurl.so`.

I am beginning to suspect the issue is that other libraries used by Julia are built with Mbed TLS rather than OpenSSL.

**Comment by mkitti** [comment]
What we might need is a version of libcurl built with Mbed TLS rather than OpenSSL.

**Comment by mkitti** [comment]
xref: https://github.com/conda-forge/julia-feedstock/pull/177

**Comment by ngam** [comment]
> xref: [conda-forge/julia-feedstock#177](https://github.com/conda-forge/julia-feedstock/pull/177)

Yep, this turns out to be a pretty difficult thing to resolve.

**Comment by ngam** [comment]
Thanks to @mkitti I think we are going to be able to resolve the issue with curl without this repopatch. So, taking curl out.

**Comment by ngam** [comment]
@conda-forge/core ready to merge, thanks

**Comment by ngam** [comment]
@conda-forge/core + @jaimergp: ready.

```diff
conda-forge-repodata-patches-feedstock$ mamba activate menv
(menv) ~/Repos/conda-forge-repodata-patches-feedstock$ python recipe/show_diff.py 
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
linux-64::julia-1.6.1-h96f571d_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h438a1c7_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-h96f571d_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.2-hbb0e23a_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h2a39549_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.3-h438a1c7_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h2a39549_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h34dcedf_2.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.4-h989b2f6_2.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.6.5-h989b2f6_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.0-h989b2f6_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h620718c_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
linux-64::julia-1.7.1-h989b2f6_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata.json.bz2
osx-64::julia-1.6.1-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-h132cb31_3.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.2-ha300950_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.3-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.4-h132cb31_2.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.6.5-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.0-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h132cb31_1.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
osx-64::julia-1.7.1-h1f75604_0.tar.bz2
-    "openlibm",
+    "openlibm <0.8.0",
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata_from_packages.json.bz2
Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata.json.bz2
```

**Review by jaimergp — APPROVED** [review]
LGTM. I'll let somebody else from @conda-forge/core do a 2nd review and merge.

**Comment by MilesCranmer** [comment]
What's the status of this? Could it be merged?

**Comment by wolfv** [comment]
sorry for the delay!

**Pull request merged**